### PR TITLE
Historical rainfall

### PR DESCRIPTION
**Pull request by t-downing**
Rainfall (IMERG) for historical cyclones by province around landfall date

### CODE REVIEW

**Comment by t-downing** [comment]
Because we are using the same code to aggregate the rainfall for both the historical storms and for Chido, I am confident that the outputs for Chido are correct. I also re-ran the plots in `exploration/landfall_rainfall.ipynb` including Chido to double-check this. For Nampula (where Chido caused the most rain), we can see it is well below even the province-level 3-yr RP threshold:

image


But just to be safe I have also sense-checked this using another source- CHIRPS. This is an imperfect comparison because we only have the CHIRPS **prelim** (which is only available pentadally) for Chido because it is so recent. And because the pentad ends on Dec 15, we miss one day of rainfall (i.e. the day after landfall, Dec 16). However we can still get a basic idea of whether Chido was super rainy or not. The plot below compares its pentadal rainfall it against other high-rainfall storms (only for Nampula, where Chido caused the most rain):

image

Basically, it looks plausible that Chido was **not** an anomalously high rainfall storm.

Note the discrepancy between CHIRPS and IMERG in absolute value- this is expected and has been noted in other contexts.

**Comment by t-downing** [comment]
Looking at the calculation of the country-wide 3-yr RP rainfall threshold from `analysis/04.2_historical_rainfall.ipynb`:

```
sorted_df = (
    combined_df.groupby("nameseason")
    .max()
    .sort_values(by="sum_mean_rain", ascending=False)
    .head(
        math.floor(23 / 3)
    )[  # 23 is the number of seasons and 3 is the target RP
        "sum_mean_rain"
    ]
    .min()
)
sorted_df
```

I think this is a sensible and quite clever way to combine the multiple admins. I suspect that Gaza wouldn't have caused any triggering with this threshold simply because it typically doesn't receive as much rain- likely due to it being somewhat more inland than e.g. Inhambane.

The only issue I could see with combining the rainfall levels this way is comparing the mean value of rainfall for areas of different size and shape, where the rainfall pattern would be somewhat concentrated (e.g. the same storm may cause a lower apparent rainfall in a large admin than a small one simply because its rainfall spike gets averaged out over a larger area). But I wouldn't worry about this here, since the admins are of pretty similar size and shape.
